File: 509 Fibonacci Number/509.js
/**
 * @param {number} N
 * @return {number}
 */
var fib = function(N) {
    let a = 0, b = 1
    for (let i = 0; i < N; ++i) {
        let c = a + b
        a = b
        b = c
    }
    return a;
};

File: 509 Fibonacci Number/509.java
class Solution {
    public int fib(int N) {
        int a = 0;
        int b = 1;
        for (int i = 0; i < N; ++i) {
            int c = a + b;
            a = b;
            b = c;
        }
        return a;
    }
}